src/kudu/consensus/consensus_queue.cc

```diff
@@ -1255,9 +1263,7 @@ int64_t PeerMessageQueue::ComputeNewWatermarkDynamicMode(int64_t* watermark) {
       continue;
     }
 
-    CHECK(peer_pb.has_permanent_uuid() && peer_pb.has_attrs() &&
```

Comment:
this check is still valid, I think although it might be happening at more appropriate places